src/System.Threading/tests/Performance/Perf.Volatile.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+
+namespace System.Threading.Tests
+{
+    public class Perf_Volatile
+    {
+        [Benchmark(InnerIterationCount = 2000)]
+        public void Read_double()
+        {
+            double location = 0;
+
+            Benchmark.Iterate(() =>
+            {
+                Volatile.Read(ref location);
```

Comment:
@stephentoub You are absolutely right, I have just ran some tests and removing the delegate approach it gets extremely cheaper. I will refactor accordingly, thanks again.